client/sync.js

```diff
@@ -163,11 +164,22 @@ const startSync = (requester) => {
       if (record.bookmark && record.bookmark.parentFolderObjectId) {
         record.bookmark.parentFolderObjectId = new Uint8Array(record.bookmark.parentFolderObjectId)
       }
-      requester.bufferedPut(categoryId, record).then(() => {
-        logSync(`sending record: ${JSON.stringify(record)}`)
-      }).catch((e) => {
-        logSync(`could not send record ${JSON.stringify(record)}: ${e}`, ERROR)
-      })
+      promisePuts.push(
+        requester.bufferedPut(categoryId, record).then(() => {
+          logSync(`sending record: ${JSON.stringify(record)}`)
+          return record
+        }).catch((e) => {
+          logSync(`could not send record ${JSON.stringify(record)}: ${e}`, ERROR)
+        })
+      )
+    })
+    Promise.all(promisePuts).then((records) => {
```

Comment:
What happens if some records from `records` failed to send? We still send the whole array in the call back.

Reply:
that would be caught in the following blocks, partial records will be sent back

Reply:
the records here is each record return from bufferedPut promises

Reply:
if this isn't clear, I can use separate array to store "SentRecords"

Reply:
maybe let's just use there something like `sentRecords` instead of `records` in the second promise? It would be less confusing in that case.

Reply:
fixed in 4ca1dbb4c2ff157f328e1637d2e0e57141dab8d7